src/components/task-item/index.js

```diff
@@ -0,0 +1,223 @@
+import css from './component.css';
+import html from './template.html';
+
+const sheet = new CSSStyleSheet();
+sheet.replaceSync(css);
+
+const DataAttributeSelector = {
+  title: 'data-title',
+  priority: 'data-priority',
+  date: 'data-date',
+  tags: 'data-tags',
+  interactive: 'interactive',
+};
+
+const UISelector = {
+  slot: 'slot',
+  details: 'details',
+  title: '.task-title',
+  priority: '.task-priority',
+  date: '.task-date',
+  tags: '.tags',
+  editBtn: '.edit-button',
+  deleteBtn: '.delete-button',
+  completeBtn: '.complete-button',
+  controlBtns: '.task-control-btn',
+};
+
+/**
+ * A custom component that displays details of a task and provides interaction
+ */
+export class TaskItem extends HTMLElement {
+  /**
+   *
+   */
+  constructor() {
+    super();
+    const shadow = this.attachShadow({ mode: 'open' });
+    shadow.adoptedStyleSheets = [sheet];
+  }
+
+  /**
+   * @returns { string[] } A list of attributes to listen to
+   */
+  static get observedAttributes() {
+    return Object.values(DataAttributeSelector); //
+  }
+
+  /**
+   * Render and add event listeners when the component is mounted to the DOM
+   */
+  connectedCallback() {
+    this.mountTemplate();
+    this.render();
+    this.addEventListeners();
+  }
+
+  /**
+   * Render the page whenever an attribute is updated
+   */
+  attributeChangedCallback() {
+    // Only update content if template is already mounted
+    if (this.shadowRoot.querySelector(UISelector.details)) {
+      this.render();
+    }
+  }
+
```

Comment:
I assume we will be needing setter methods along with getter methods (which you have implemented)